stablehlo/tests/stablehlo_aggressive_simplification.mlir

```diff
@@ -664,6 +664,277 @@ func.func @reduce_zero_ext(%arg0: tensor<0xi1>) -> tensor<i32> {
 
 // -----
 
+// Each reduce_unused_* test case is accompanied by an ASCII diagram that
+// represents the surveyed reduce operation in a compact form:
+//
+//  (a, a1) (b, b1)
+//    ▴       ▴
+//    │       │
+//    │       │
+//    │       │
+//    │       │
+//    r0      r1
+//    U
+//
+// In this case:
+// a,  b  - are the operands to be reduced, i.e. %arg0, %arg1.
+// a1, b1 - are the initial values of the operands, i.e. %0, %1.
+// r0, r1 - are the results of stablehlo.reduce and/or stablehlo.return operation
+//          (they are equivalent in this context), i.e %2#0, %2#1 and %3, %4.
+// Arrows show which results depend on which inputs.
+// U below the result means it is used (live).
+//
+// To drop r1 use pair (b, b1).
+
+// CHECK-LABEL: func.func @reduce_unused_case0
+func.func @reduce_unused_case0(%arg0: tensor<8xi64>,
```

Comment:
Nit: consider `CHECK-SAME: %[[ARG_0:.+]]: tensor<8xi64>` to avoid depending on the name of the operand (see other examples above in this file; similar comment applies to other cases below).